File: frontend/src/types/question.ts
export type Question = {
  id: string;
  type: "MULTIPLE CHOICE" | "OPEN ENDED";
  question_text: string;
  options: Record<string, string> | null;
  correct_answer: string | null;
  explanation: string;
  version: string;
  subject: string;
  topic: string;
};

export type QuestionType = "mcq" | "open";
export type SourceType = "similarity" | "document";

export enum QuestionTypeMapping {
  mcq = "MULTIPLE CHOICE",
  open = "OPEN ENDED",
}

export enum SourceTypeMapping {
  document = "PDF WORKSPACE",
  similarity = "SIMILAR QUESTIONS",
}
export type QuestionContent = {
  question_type: QuestionType;
  question_text: string;
  options: Record<string, string> | null;
  correct_answer: string;
  explanation: string;
  tags?: Record<string, unknown> | null;
  confidence_score?: number | null;
};

export type RefinementResponse = {
  question_id: string;
  version: number;
  question: QuestionContent;
};

export type RecentQuestion = {
  session_id: string;
  question_type: QuestionType;
  source_type: SourceType;
  quantity: number;
  created_at: string;
};

File: backend/app/models/question.py
from datetime import datetime
from typing import Any, Literal, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class Question(BaseModel):
    id: UUID
    user_id: UUID
    session_id: UUID
    document_id: UUID
    source_type: Literal["document", "similarity"] = "document"
    question_type: Literal["mcq", "open"]
    question_text: str
    options: Optional[dict[str, Any]] = None
    correct_answer: str
    explanation: str
    tags: Optional[dict[str, Any]] = None
    confidence_score: Optional[float] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
